### PR TITLE
ROCm 5.2.0 Updates - File/Folder Reorg Changes

### PR DESCRIPTION
**Pull request by arvindcheru**
- Updated HIP Path additions
- Package file/folder reorg applied for lib (ROCMPATH/lib) (removed component name prefix)
- Package file/folder reorg applied for include (ROCMPATH/include/<mivisionx>)
- Package file/folder reorg applied for docs, samples, test binaries.

### CODE REVIEW

**Comment by arvindcheru** [comment]
@asalmanp, @kiritigowda  Kindly share your input/comments if there are any

**Review by asalmanp — CHANGES_REQUESTED** [review]
@arvindcheru this PR introduces lots of cmake warnings (i.e., ROCMChecks WARNING). please take a look and fix these warnings (i.e., just run cmake ..)

**Comment by arvindcheru** [comment]
> @arvindcheru this PR introduces lots of cmake warnings (i.e., ROCMChecks WARNING see this [link ](http://math-ci.rocm.amd.com/blue/rest/organizations/jenkins/pipelines/compute-rocm-dkms-no-npi-hipclang/pipelines/Short-GPUOpen/pipelines/MIVisionX/branches/PR-786/runs/3/nodes/191/log/?start=0)for the log file from the CI). please take a look and fix these warnings (i.e., just run cmake ..)

At first look, This is warning generated by latest rocm-cmake on existing code since CMAKE_CXX_COMPILER is set in the code. Like set(CMAKE_CXX_COMPILER ${COMPILER_FOR_HIP}) in amd_openvx/openvx/hipvx/CMakeLists.txt line 36 for example.

**Comment by paveltc** [comment]
@asalmanp I have built both the HIP and OCL backends with this PR and run some of the sanity tests. It built and ran the tests fine.

**Review by asalmanp — CHANGES_REQUESTED** [review]

File: amd_openvx/openvx/CMakeLists.txt [hunk 1]
```diff
@@ -183,3 +183,25 @@ else()
     target_link_libraries(openvx dl m)
     target_link_libraries(vxu openvx)
 endif()
+
+if(BUILD_FILE_REORG_BACKWARD_COMPATIBILITY)
+  include(ROCMHeaderWrapper)
+  # Copy Header files to build directory
+  configure_file("${PROJECT_SOURCE_DIR}/include/vx_ext_amd.h" "${PROJECT_BINARY_DIR}/include/mivisionx/vx_ext_amd.h" COPYONLY)
```
Comment (asalmanp) [inline amd_openvx/openvx/CMakeLists.txt:190]: change ${PROJECT_BINARY_DIR}/include/mivisionx/vx_ext_amd.h" to ${PROJECT_BINARY_DIR}/include/vx_ext_amd.h"

File: amd_openvx_extensions/amd_loomsl/CMakeLists.txt [hunk 2]
```diff
@@ -55,11 +55,25 @@ add_library(vx_loomsl SHARED ${SOURCES})
...
 	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MD")
 	set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /MDd")
 else()
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.2 -std=c++11")
 endif()
+
+if(BUILD_FILE_REORG_BACKWARD_COMPATIBILITY)
+  #include(ROCMHeaderWrapper)
+  # Copy Header files to build directory
+  configure_file("live_stitch_api.h" "${PROJECT_BINARY_DIR}/include/mivisionx/live_stitch_api.h" COPYONLY)
```
Comment (asalmanp) [inline amd_openvx_extensions/amd_loomsl/CMakeLists.txt:70]: change "${PROJECT_BINARY_DIR}/include/mivisionx/live_stitch_api.h" to "${PROJECT_BINARY_DIR}/include/live_stitch_api.h"

File: amd_openvx_extensions/amd_media/CMakeLists.txt [hunk 3]
```diff
@@ -88,11 +88,25 @@ elseif (GPU_SUPPORT AND "${BACKEND}" STREQUAL "HIP")
...
     set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")
     set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /MTd")
 else()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations -msse4.2 -std=c++11")
 endif()
+
+if(BUILD_FILE_REORG_BACKWARD_COMPATIBILITY)
+  #include(ROCMHeaderWrapper)
+  # Copy Header files to build directory
+  configure_file("include/vx_amd_media.h" "${PROJECT_BINARY_DIR}/include/mivisionx/vx_amd_media.h" COPYONLY)
```
Comment (asalmanp) [inline amd_openvx_extensions/amd_media/CMakeLists.txt:103]: change "${PROJECT_BINARY_DIR}/include/mivisionx/vx_amd_media.h" to
"${PROJECT_BINARY_DIR}/include/vx_amd_media.h"

File: amd_openvx_extensions/amd_opencv/CMakeLists.txt [hunk 4]
```diff
@@ -117,3 +117,18 @@ else()
 	target_link_libraries(vx_opencv openvx)
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdata-sections")
 endif()
+
+
+if(BUILD_FILE_REORG_BACKWARD_COMPATIBILITY)
+  #include(ROCMHeaderWrapper)
+  # Copy Header files to build directory
+  configure_file("include/vx_ext_opencv.h" "${PROJECT_BINARY_DIR}/include/mivisionx/vx_ext_opencv.h" COPYONLY)
```
Comment (asalmanp) [inline amd_openvx_extensions/amd_opencv/CMakeLists.txt:125]: change "${PROJECT_BINARY_DIR}/include/mivisionx/vx_ext_opencv.h" to
"${PROJECT_BINARY_DIR}/include/vx_ext_opencv.h"

File: utilities/CMakeLists.txt [hunk 5]
```diff
@@ -86,3 +86,19 @@ else()
     message("-- ${Green}Utilities -- runvx module added with CPU support${ColourReset}")
     message("-- ${Red}WARNING: Utilities -- runcl, loom_shell, & mv_deploy modules excluded${ColourReset}")
 endif()
+
+if(BUILD_FILE_REORG_BACKWARD_COMPATIBILITY)
+  if(FFMPREG_FOUND)
+    #include(ROCMHeaderWrapper)
+    # Copy Header files to build directory
+    configure_file("./mv_deploy/mvdeploy_api.h" "${PROJECT_BINARY_DIR}/include/mivisionx/mvdeploy_api.h" COPYONLY)
```
Comment (asalmanp) [inline utilities/CMakeLists.txt:94]: change "${PROJECT_BINARY_DIR}/include/mivisionx/mvdeploy_api.h" to "${PROJECT_BINARY_DIR}/include/mvdeploy_api.h"

**Review by asalmanp — CHANGES_REQUESTED** [review]

**Comment by kiritigowda** [comment]
add_definitions(-w) — to suppress warnings @arvindcheru?

**Comment by arvindcheru** [comment]
> @asalmanp I tried to build the bubble pop app and got the following error: 14 | #warning "This file is deprecated. Use the header file from /opt/rocm/include/mivisionx/vx.h by using #include <mivisionx/vx.h>" | ^~~~~~~ /opt/rocm/mivisionx/include/VX/vx.h:16:10: fatal error: ../../../include/mivisionx/vx.h: No such file or directory 16 | #include "../../../include/mivisionx/vx.h" | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Thanks for comments, at first looks the relative path generated in wrapper need to be analyzed. Will check this and update shortly

**Comment by kiritigowda** [comment]
@arvindcheru update on warnings. The current TOT with rocm-cmake master does not throw warnings

```
cmake ../MIVisionX/
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- MIVisionX Version -- 2.1.0
-- MIVisionX Install Path -- /opt/rocm/mivisionx
-- MIVisionX Backend set to -- HIP
-- MIVisionX Build Type -- Release
-- MIVisionX Developer Options
--      -D NEURAL_NET=OFF [Turn OFF Neural Net Modules (default:ON)]
--      -D ROCAL=OFF [Turn OFF ROCAL Modules (default:ON)]
--      -D LOOM=OFF [Turn OFF LOOM Modules (default:ON)]
--      -D GPU_SUPPORT=OFF [Turn OFF GPU support (default:ON)]
--      -D BACKEND=OPENCL [select OPENCL for GPU backend [options:CPU/OPENCL/HIP](default:HIP)]
-- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37
-- AMD OpenVX -- OpenVX built with HIP Support
-- AMD OpenVX -- OpenVX module added
-- OpenCL Found - /opt/rocm/lib/libOpenCL.so
-- ROCm OpenCL Found - Force OpenCL_LIBRARIES & OpenCL_INCLUDE_DIRS to use ROCm OpenCL
-- OpenCL - Setting CL_TARGET_OPENCL_VERSION=220
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'libavcodec'
--   Found libavcodec, version 58.18.100
-- Checking for module 'libavformat'
--   Found libavformat, version 58.12.100
-- Checking for module 'libavutil'
--   Found libavutil, version 56.14.100
-- Using FFMPEG -- Libraries:/usr/local/lib/libavcodec.so;/usr/local/lib/libavformat.so;/usr/local/lib/libavutil.so;/usr/local/lib/libswscale.so Includes:/usr/local/include
-- amd_media -- Building with HIP
-- AMD OpenVX Media Extension -- amd_media module added
-- WARNING: MIOPEN with HIP backend Not Found -- amd_nn module excluded
-- WARNING: RPP with the HIP backend Not Found (HINT: run setup script with the HIP backend first!) -- amd_rpp module excluded.
-- OpenCV Found -- Version-4.5.X Supported
-- AMD OpenVX OpenCV Extension -- amd_opencv module added
-- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37
-- runvx built with HIP Support
-- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37
-- Utilities -- runvx and mv_deploy modules added
-- Found HIP for rocAL: 5.0.13601-6b731c37/opt/rocm/hip
-- WARNING: RPP with the HIP backend Not Found (HINT: run setup script with the HIP backend first!) -- rocAL_rpp module excluded.
-- WARNING: rocAL module excluded 
-- Configuring done
-- Generating done
-- Build files have been written to: /workspace/build
```

**Comment by kiritigowda** [comment]
@asalmanp -- the warnings only come up with this PR, see comment above.
```
cmake ../MIVisionX/
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- MIVisionX Version -- 2.1.0
-- MIVisionX Install Path -- /opt/rocm
-- MIVisionX Backend set to -- HIP
-- MIVisionX Build Type -- Release
-- MIVisionX Developer Options
--      -D NEURAL_NET=OFF [Turn OFF Neural Net Modules (default:ON)]
--      -D ROCAL=OFF [Turn OFF ROCAL Modules (default:ON)]
--      -D LOOM=OFF [Turn OFF LOOM Modules (default:ON)]
--      -D GPU_SUPPORT=OFF [Turn OFF GPU support (default:ON)]
--      -D BACKEND=OPENCL [select OPENCL for GPU backend [options:CPU/OPENCL/HIP](default:HIP)]
-- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37

*******************************************************************************
*------------------------------- ROCMChecks WARNING --------------------------*
  Options and properties should be set on a cmake target where possible. The
  variable 'CMAKE_CXX_COMPILER' may be set by the cmake toolchain, either by
  calling 'cmake -DCMAKE_CXX_COMPILER="/opt/rocm/bin/hipcc"'
  or set in a toolchain file and added with
  'cmake -DCMAKE_TOOLCHAIN_FILE=<toolchain-file>'. ROCMChecks now calling:
CMake Warning at /opt/rocm-5.0.0/share/rocm/cmake/ROCMChecks.cmake:46 (message):
  'CMAKE_CXX_COMPILER' is set at
  /workspace/MIVisionX/amd_openvx/openvx/hipvx/CMakeLists.txt:<line#> shown
  below:
Call Stack (most recent call first):
  amd_openvx/openvx/hipvx/CMakeLists.txt:9999 (rocm_check_toolchain_var)
  amd_openvx/openvx/hipvx/CMakeLists.txt:36 (set)


*-----------------------------------------------------------------------------*
*******************************************************************************


*******************************************************************************
*------------------------------- ROCMChecks WARNING --------------------------*
  Options and properties should be set on a cmake target where possible. The
  variable 'CMAKE_CXX_FLAGS' may be set by the cmake toolchain, either by
  calling 'cmake -DCMAKE_CXX_FLAGS=" -O3 "'
  or set in a toolchain file and added with
  'cmake -DCMAKE_TOOLCHAIN_FILE=<toolchain-file>'. ROCMChecks now calling:
CMake Warning at /opt/rocm-5.0.0/share/rocm/cmake/ROCMChecks.cmake:46 (message):
  'CMAKE_CXX_FLAGS' is set at
  /workspace/MIVisionX/amd_openvx/openvx/hipvx/CMakeLists.txt:<line#> shown
  below:
Call Stack (most recent call first):
  amd_openvx/openvx/hipvx/CMakeLists.txt:9999 (rocm_check_toolchain_var)
  amd_openvx/openvx/hipvx/CMakeLists.txt:37 (set)


*-----------------------------------------------------------------------------*
*******************************************************************************

-- AMD OpenVX -- OpenVX built with HIP Support

*******************************************************************************
*------------------------------- ROCMChecks WARNING --------------------------*
  Options and properties should be set on a cmake target where possible. The
  variable 'CMAKE_CXX_FLAGS' may be set by the cmake toolchain, either by
  calling 'cmake -DCMAKE_CXX_FLAGS=" -O3 -msse4.2 -std=gnu++14"'
  or set in a toolchain file and added with
  'cmake -DCMAKE_TOOLCHAIN_FILE=<toolchain-file>'. ROCMChecks now calling:
CMake Warning at /opt/rocm-5.0.0/share/rocm/cmake/ROCMChecks.cmake:46 (message):
  'CMAKE_CXX_FLAGS' is set at
  /workspace/MIVisionX/amd_openvx/openvx/CMakeLists.txt:<line#> shown below:
Call Stack (most recent call first):
  amd_openvx/openvx/CMakeLists.txt:9999 (rocm_check_toolchain_var)
  amd_openvx/openvx/CMakeLists.txt:182 (set)


*-----------------------------------------------------------------------------*
*******************************************************************************

-- AMD OpenVX -- OpenVX module added
-- OpenCL Found - /opt/rocm/lib/libOpenCL.so
-- ROCm OpenCL Found - Force OpenCL_LIBRARIES & OpenCL_INCLUDE_DIRS to use ROCm OpenCL
-- OpenCL - Setting CL_TARGET_OPENCL_VERSION=220
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'libavcodec'
--   Found libavcodec, version 58.18.100
-- Checking for module 'libavformat'
--   Found libavformat, version 58.12.100
-- Checking for module 'libavutil'
--   Found libavutil, version 56.14.100
-- Using FFMPEG -- Libraries:/usr/local/lib/libavcodec.so;/usr/local/lib/libavformat.so;/usr/local/lib/libavutil.so;/usr/local/lib/libswscale.so Includes:/usr/local/include
-- amd_media -- Building with HIP

*******************************************************************************
*------------------------------- ROCMChecks WARNING --------------------------*
  Options and properties should be set on a cmake target where possible. The
  variable 'CMAKE_CXX_FLAGS' may be set by the cmake toolchain, either by
  calling 'cmake -DCMAKE_CXX_FLAGS=" -O3 -Wno-deprecated-declarations -msse4.2 -std=c++11"'
  or set in a toolchain file and added with
  'cmake -DCMAKE_TOOLCHAIN_FILE=<toolchain-file>'. ROCMChecks now calling:
CMake Warning at /opt/rocm-5.0.0/share/rocm/cmake/ROCMChecks.cmake:46 (message):
  'CMAKE_CXX_FLAGS' is set at
  /workspace/MIVisionX/amd_openvx_extensions/amd_media/CMakeLists.txt:<line#>
  shown below:
Call Stack (most recent call first):
  amd_openvx_extensions/amd_media/CMakeLists.txt:9999 (rocm_check_toolchain_var)
  amd_openvx_extensions/amd_media/CMakeLists.txt:97 (set)


*-----------------------------------------------------------------------------*
*******************************************************************************

-- AMD OpenVX Media Extension -- amd_media module added
-- WARNING: MIOPEN with HIP backend Not Found -- amd_nn module excluded
-- WARNING: RPP with the HIP backend Not Found (HINT: run setup script with the HIP backend first!) -- amd_rpp module excluded.
-- OpenCV Found -- Version-4.5.X Supported

*******************************************************************************
*------------------------------- ROCMChecks WARNING --------------------------*
  Options and properties should be set on a cmake target where possible. The
  variable 'CMAKE_CXX_FLAGS' may be set by the cmake toolchain, either by
  calling 'cmake -DCMAKE_CXX_FLAGS=" -O3 -fdata-sections"'
  or set in a toolchain file and added with
  'cmake -DCMAKE_TOOLCHAIN_FILE=<toolchain-file>'. ROCMChecks now calling:
CMake Warning at /opt/rocm-5.0.0/share/rocm/cmake/ROCMChecks.cmake:46 (message):
  'CMAKE_CXX_FLAGS' is set at
  /workspace/MIVisionX/amd_openvx_extensions/amd_opencv/CMakeLists.txt:<line#>
  shown below:
Call Stack (most recent call first):
  amd_openvx_extensions/amd_opencv/CMakeLists.txt:9999 (rocm_check_toolchain_var)
  amd_openvx_extensions/amd_opencv/CMakeLists.txt:118 (set)


*-----------------------------------------------------------------------------*
*******************************************************************************

-- AMD OpenVX OpenCV Extension -- amd_opencv module added
-- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37
-- runvx built with HIP Support

*******************************************************************************
*------------------------------- ROCMChecks WARNING --------------------------*
  Options and properties should be set on a cmake target where possible. The
  variable 'CMAKE_CXX_FLAGS' may be set by the cmake toolchain, either by
  calling 'cmake -DCMAKE_CXX_FLAGS=" -O3 -std=c++11"'
  or set in a toolchain file and added with
  'cmake -DCMAKE_TOOLCHAIN_FILE=<toolchain-file>'. ROCMChecks now calling:
CMake Warning at /opt/rocm-5.0.0/share/rocm/cmake/ROCMChecks.cmake:46 (message):
  'CMAKE_CXX_FLAGS' is set at
  /workspace/MIVisionX/utilities/runvx/CMakeLists.txt:<line#> shown below:
Call Stack (most recent call first):
  utilities/runvx/CMakeLists.txt:9999 (rocm_check_toolchain_var)
  utilities/runvx/CMakeLists.txt:134 (set)


*-----------------------------------------------------------------------------*
*******************************************************************************


*******************************************************************************
*------------------------------- ROCMChecks WARNING --------------------------*
  Options and properties should be set on a cmake target where possible. The
  variable 'CMAKE_CXX_FLAGS' may be set by the cmake toolchain, either by
  calling 'cmake -DCMAKE_CXX_FLAGS=" -O3 -std=c++11 -DHAVE_OpenSSL=1"'
  or set in a toolchain file and added with
  'cmake -DCMAKE_TOOLCHAIN_FILE=<toolchain-file>'. ROCMChecks now calling:
CMake Warning at /opt/rocm-5.0.0/share/rocm/cmake/ROCMChecks.cmake:46 (message):
  'CMAKE_CXX_FLAGS' is set at
  /workspace/MIVisionX/utilities/runvx/CMakeLists.txt:<line#> shown below:
Call Stack (most recent call first):
  utilities/runvx/CMakeLists.txt:9999 (rocm_check_toolchain_var)
  utilities/runvx/CMakeLists.txt:141 (set)


*-----------------------------------------------------------------------------*
*******************************************************************************

-- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37

*******************************************************************************
*------------------------------- ROCMChecks WARNING --------------------------*
  Options and properties should be set on a cmake target where possible. The
  variable 'CMAKE_CXX_FLAGS' may be set by the cmake toolchain, either by
  calling 'cmake -DCMAKE_CXX_FLAGS=" -O3 -std=c++11 -msse4.1 -mf16c"'
  or set in a toolchain file and added with
  'cmake -DCMAKE_TOOLCHAIN_FILE=<toolchain-file>'. ROCMChecks now calling:
CMake Warning at /opt/rocm-5.0.0/share/rocm/cmake/ROCMChecks.cmake:46 (message):
  'CMAKE_CXX_FLAGS' is set at
  /workspace/MIVisionX/utilities/mv_deploy/CMakeLists.txt:<line#> shown
  below:
Call Stack (most recent call first):
  utilities/mv_deploy/CMakeLists.txt:9999 (rocm_check_toolchain_var)
  utilities/mv_deploy/CMakeLists.txt:35 (set)


*-----------------------------------------------------------------------------*
*******************************************************************************

-- Utilities -- runvx and mv_deploy modules added
-- Found HIP for rocAL: 5.0.13601-6b731c37/opt/rocm/hip
-- WARNING: RPP with the HIP backend Not Found (HINT: run setup script with the HIP backend first!) -- rocAL_rpp module excluded.
-- WARNING: rocAL module excluded 
-- FILE_REORG - Install wrapper headers/Symlink files
-- Configuring done
-- Generating done
-- Build files have been written to: /workspace/build
```

**Comment by arvindcheru** [comment]
> @arvindcheru update on warnings. The current TOT with rocm-cmake master does not throw warnings
> 
> ```
> cmake ../MIVisionX/
> -- The C compiler identification is GNU 9.4.0
> -- The CXX compiler identification is GNU 9.4.0
> -- Check for working C compiler: /usr/bin/cc
> -- Check for working C compiler: /usr/bin/cc -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Check for working CXX compiler: /usr/bin/c++
> -- Check for working CXX compiler: /usr/bin/c++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- MIVisionX Version -- 2.1.0
> -- MIVisionX Install Path -- /opt/rocm/mivisionx
> -- MIVisionX Backend set to -- HIP
> -- MIVisionX Build Type -- Release
> -- MIVisionX Developer Options
> --      -D NEURAL_NET=OFF [Turn OFF Neural Net Modules (default:ON)]
> --      -D ROCAL=OFF [Turn OFF ROCAL Modules (default:ON)]
> --      -D LOOM=OFF [Turn OFF LOOM Modules (default:ON)]
> --      -D GPU_SUPPORT=OFF [Turn OFF GPU support (default:ON)]
> --      -D BACKEND=OPENCL [select OPENCL for GPU backend [options:CPU/OPENCL/HIP](default:HIP)]
> -- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37
> -- AMD OpenVX -- OpenVX built with HIP Support
> -- AMD OpenVX -- OpenVX module added
> -- OpenCL Found - /opt/rocm/lib/libOpenCL.so
> -- ROCm OpenCL Found - Force OpenCL_LIBRARIES & OpenCL_INCLUDE_DIRS to use ROCm OpenCL
> -- OpenCL - Setting CL_TARGET_OPENCL_VERSION=220
> -- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
> -- Checking for module 'libavcodec'
> --   Found libavcodec, version 58.18.100
> -- Checking for module 'libavformat'
> --   Found libavformat, version 58.12.100
> -- Checking for module 'libavutil'
> --   Found libavutil, version 56.14.100
> -- Using FFMPEG -- Libraries:/usr/local/lib/libavcodec.so;/usr/local/lib/libavformat.so;/usr/local/lib/libavutil.so;/usr/local/lib/libswscale.so Includes:/usr/local/include
> -- amd_media -- Building with HIP
> -- AMD OpenVX Media Extension -- amd_media module added
> -- WARNING: MIOPEN with HIP backend Not Found -- amd_nn module excluded
> -- WARNING: RPP with the HIP backend Not Found (HINT: run setup script with the HIP backend first!) -- amd_rpp module excluded.
> -- OpenCV Found -- Version-4.5.X Supported
> -- AMD OpenVX OpenCV Extension -- amd_opencv module added
> -- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37
> -- runvx built with HIP Support
> -- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37
> -- Utilities -- runvx and mv_deploy modules added
> -- Found HIP for rocAL: 5.0.13601-6b731c37/opt/rocm/hip
> -- WARNING: RPP with the HIP backend Not Found (HINT: run setup script with the HIP backend first!) -- rocAL_rpp module excluded.
> -- WARNING: rocAL module excluded 
> -- Configuring done
> -- Generating done
> -- Build files have been written to: /workspace/build
> ```

Ok Thanks Kiriti, Did the MIVisionX latest version updated to latest rocm-cmake? So shall I remove all rocm-cmake download code from here?

**Comment by arvindcheru** [comment]
> 

As you can see the line number of these warnings - They are not from new PR changes.

**Comment by kiritigowda** [comment]
@arvindcheru needs to sync with develop.

**Comment by arvindcheru** [comment]
> @arvindcheru needs to sync with develop.

I think we are comparing different rocm-cmake version
Kindly take a look on rocm-cmake version with "https://github.com/RadeonOpenCompute/rocm-cmake/blob/master/share/rocm/cmake/ROCMHeaderWrapper.cmake" file in it. this is in master rocm-cmake. Please check whether the latest version is with matching this version. I see it is not matching.

**Comment by kiritigowda** [comment]
@arvindcheru now the develop branch uses master branch for rocm-cmake. Below is the log. Let me know if you can try this with the develop branch.

```
cmake ../MIVisionX/
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- MIVisionX Version -- 2.1.0
-- MIVisionX Install Path -- /opt/rocm/mivisionx
-- MIVisionX Backend set to -- HIP
-- MIVisionX Build Type -- Release
-- MIVisionX Developer Options
--      -D NEURAL_NET=OFF [Turn OFF Neural Net Modules (default:ON)]
--      -D ROCAL=OFF [Turn OFF ROCAL Modules (default:ON)]
--      -D LOOM=OFF [Turn OFF LOOM Modules (default:ON)]
--      -D GPU_SUPPORT=OFF [Turn OFF GPU support (default:ON)]
--      -D BACKEND=OPENCL [select OPENCL for GPU backend [options:CPU/OPENCL/HIP](default:HIP)]
-- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37
-- AMD OpenVX -- OpenVX built with HIP Support
-- AMD OpenVX -- OpenVX module added
-- hip::amdhip64 is SHARED_LIBRARY
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'libavcodec'
--   Found libavcodec, version 58.18.100
-- Checking for module 'libavformat'
--   Found libavformat, version 58.12.100
-- Checking for module 'libavutil'
--   Found libavutil, version 56.14.100
-- Using FFMPEG -- Libraries:/usr/local/lib/libavcodec.so;/usr/local/lib/libavformat.so;/usr/local/lib/libavutil.so;/usr/local/lib/libswscale.so Includes:/usr/local/include
-- amd_media -- Building with HIP
-- AMD OpenVX Media Extension -- amd_media module added
-- Found Threads: TRUE  
-- hip::amdhip64 is SHARED_LIBRARY
-- hip::amdhip64 is SHARED_LIBRARY
-- Found Protobuf: /usr/local/lib/libprotobuf.so;-lpthread (found version "3.12.0") 
-- Inference Generator -- utilities/inference_generator module added
-- amd_nn -- Building with HIP
-- AMD OpenVX Neural Network Extension -- amd_nn module added with HIP backend
-- Found AMDRPP: /opt/rocm/rpp/lib/libamd_rpp.so  
-- amd_rpp -- Building with HIP
-- AMD Radeon Performance Primitives -- amd_rpp module added with the HIP backend support
-- OpenCV Found -- Version-4.5.X Supported
-- AMD OpenVX OpenCV Extension -- amd_opencv module added
-- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37
-- runvx built with HIP Support
-- Found HIP: /opt/rocm/hip version: 5.0.13601-6b731c37
-- Utilities -- runvx and mv_deploy modules added
-- Found HIP for rocAL: 5.0.13601-6b731c37/opt/rocm/hip
-- ROCm Augmentation Library -- rocAL module added
-- rocAL library is going to be built with HIP support 
-- rocAL built with ENABLE_SIMD
-- rocAL - CMAKE_CXX_FLAGS: -O3 -O3 -fopenmp -msse4.2 -mavx2 -Wall  -fPIC -pg -pthread -std=gnu++14 -Wno-deprecated-declarations
-- rocAL library is going to be built with video decode functionality 
-- Configuring done
-- Generating done
-- Build files have been written to: /workspace/build
```

**Comment by arvindcheru** [comment]
> @asalmanp I tried to build the bubble pop app and got the following error: 14 | #warning "This file is deprecated. Use the header file from /opt/rocm/include/mivisionx/vx.h by using #include <mivisionx/vx.h>" | ^~~~~~~ /opt/rocm/mivisionx/include/VX/vx.h:16:10: fatal error: ../../../include/mivisionx/vx.h: No such file or directory 16 | #include "../../../include/mivisionx/vx.h" | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Just to avoid confusion, quoting here the latest update on this - should be addressed with this commit  https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pull/786/commits/a8afcef76362f10fa58c5250a90dc8531097a105

**Comment by paveltc** [comment]
@asalmanp @arvindcheru I can now build bubble pop. The error no longer appears. I can't test it however because I am building it on a remote system, which is all that I have available.

**Comment by kiritigowda** [comment]
@arvindcheru updates to MIGraphX CMake needed - https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/849679ad6ea7fb2cfbd2f4225063a7e36d2aef8f/amd_openvx_extensions/amd_migraphx/CMakeLists.txt#L21

**Comment by arvindcheru** [comment]
> @arvindcheru updates to MIGraphX CMake needed -
> 
> https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/849679ad6ea7fb2cfbd2f4225063a7e36d2aef8f/amd_openvx_extensions/amd_migraphx/CMakeLists.txt#L21

@kiritigowda , yes these new files being updated now, will be merging shortly

**Comment by kiritigowda** [comment]
@asalmanp  to verify AMD_MIGraphX

**Comment by asalmanp** [comment]
> @asalmanp to verify AMD_MIGraphX

@kiritigowda both mnist and resnet50 migraphx node tests are building and running fine with this PR.

**Review by asalmanp — APPROVED** [review]

**Pull request merged**